Sources/StreamChat/APIClient/Endpoints/EndpointPath+OfflineRequest.swift

```diff
@@ -7,14 +7,15 @@ import Foundation
 extension EndpointPath {
     var shouldBeQueuedOffline: Bool {
         switch self {
-        case .sendMessage, .editMessage, .deleteMessage, .pinMessage, .unpinMessage, .addReaction, .deleteReaction:
+        case .editMessage, .deleteMessage, .pinMessage, .unpinMessage, .addReaction, .deleteReaction:
             return true
         case .createChannel, .connect, .sync, .users, .guest, .members, .search, .devices, .channels, .updateChannel,
              .deleteChannel, .channelUpdate, .muteChannel, .showChannel, .truncateChannel, .markChannelRead, .markChannelUnread,
              .markAllChannelsRead, .channelEvent, .stopWatchingChannel, .pinnedMessages, .uploadAttachment, .message,
              .replies, .reactions, .messageAction, .banMember, .flagUser, .flagMessage, .muteUser, .translateMessage,
              .callToken, .createCall, .deleteFile, .deleteImage, .og, .appSettings, .threads, .thread, .markThreadRead, .markThreadUnread, .polls, .pollsQuery,
-             .poll, .pollOption, .pollOptions, .pollVotes, .pollVoteInMessage, .pollVote, .unread, .blockUser, .unblockUser:
+             .poll, .pollOption, .pollOptions, .pollVotes, .pollVoteInMessage, .pollVote, .unread, .blockUser, .unblockUser,
+             .sendMessage:
```

Comment:
why is this?

Reply:
APIClient sends failed API requests to SyncRepository [here](https://github.com/GetStream/stream-chat-swift/blob/8a9d94f7ed7b9b25577a9f181c4ab881bc64fbc2/Sources/StreamChat/APIClient/APIClient.swift#L171). This will then be passed to `OfflineRequestsRepository` which uses this property [for deciding if it should retry](https://github.com/GetStream/stream-chat-swift/blob/develop/Sources/StreamChat/Repositories/OfflineRequestsRepository.swift#L164) the failed request.

If we allow this and send message fails while there are multiple messages queued for sending, then it causes the race between offline repo retrying a message and message sender trying to send a next message which leads to incorrect message order.

I'll add it to the PR description.